pageserver/src/tenant/timeline/layer_manager.rs

```diff
@@ -200,7 +200,34 @@ impl LayerManager {
             metrics.record_new_file_metrics(l.layer_desc().file_size);
         }
         for l in compact_from {
-            Self::delete_historic_layer(l, &mut updates, &mut self.layer_fmgr);
+            Self::delete_historic_layer(l, &mut updates, &mut self.layer_fmgr, true);
+        }
+        updates.flush();
+    }
+
+    /// Called when compaction is completed.
+    pub(crate) fn rewrite_layers(
+        &mut self,
+        rewrite_layers: &[(Layer, ResidentLayer)],
+        drop_layers: &[Layer],
+        metrics: &TimelineMetrics,
+    ) {
+        let mut updates = self.layer_map.batch_update();
+        for (old_layer, new_layer) in rewrite_layers {
+            debug_assert!(old_layer.layer_desc().key_range == new_layer.layer_desc().key_range);
+            debug_assert!(old_layer.layer_desc().lsn_range == new_layer.layer_desc().lsn_range);
```

Comment:
Much better to use 
```suggestion
            debug_assert_eq!(old_layer.layer_desc().key_range, new_layer.layer_desc().key_range);
            debug_assert_eq!(old_layer.layer_desc().lsn_range, new_layer.layer_desc().lsn_range);
```

As the assertion message will otherwise contain just "assertion failed", or something like that.

Reply:
Will pull into #7531 (I just dropped these lines in this PR 6a995dd43)